.utils/terraform_lock_precommit.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+# Find all .terraform.lock.hcl files that have been modified or added to the staging
+staged_lock_files=$(git status --porcelain | grep '[AM][ ]*s.*.terraform.lock.hcl$' | awk '{print $2}')
+
+if [[ -z "$staged_lock_files" ]]; then
+  echo "No .terraform.lock.hcl files to process."
+  exit 0
+fi
+
+# Run terraform providers lock in each directory containing a .terraform.lock.hcl file
+for lockfile in $staged_lock_files; do
+  dir=$(dirname "$lockfile")
+
+  echo "Run terraform providers lock in: $dir"
+
+  # Go to the directory
+  cd "$dir" || exit
+
+  # Run terraform providers lock
+  terraform providers lock \
+    -platform=windows_amd64 \
+    -platform=darwin_amd64 \
+    -platform=darwin_arm64 \
+    -platform=linux_amd64
+
+  # Return to the previous directory
+  cd - > /dev/null
+done
```

Comment:
Can we move this script in dx and use it as a pre-commit in the other repos?

Reply:
Pretty much as it's been done for the script that checks the modules lock file
https://github.com/pagopa/dx/blob/main/.pre-commit-hooks.yaml

Reply:
this is a great idea! I'll try to implement it

Reply:
done!